src/common.h

```diff
@@ -16,6 +16,12 @@
 
 #include <math.h>
 
+#if __GNUC__ >= 4 /* also works for Clang */
```

Comment:
You must check for __GNUC__ being defined and also exclude windows here.
Something like:
`#if !defined(_WIN32) && (defined(__GNUC__) && __GNUC__ >= 4) /* also works for Clang */`

Reply:
in general, you dont need to check if a macro is defined before using it. if `__GNUC__` isn't defined, the clause will be false. The Win32 stuff is correct though.